Fix CI for staging deployment

aaronxsu opened the pull request:
## Overview

This PR:
- updates the `infra` script to read in AWS creds related env vars
- updates `docker-compose` file used to update job def of batch jobs to pass in the env vars


### Demo

See the green checkmark of the following GH action job run:

https://github.com/azavea/pfb-network-connectivity/actions/runs/13292649266

Staging ECS tasks were cycled through earlier today

<img width="1555" alt="Screenshot 2025-02-12 at 3 58 07 PM" src="https://github.com/user-attachments/assets/aa326775-bb97-451e-9b83-9e10b4e13a37" />

## Testing Instructions

 * Push up a `test` branch and let the GH actions pick up the job
 * The job should succeed
 * A staging deployment should trigger


## Checklist

- [ ] Add entry to CHANGELOG.md

Resolves #963

aaronxsu commented on .github/workflows/ci.yml:
```diff
@@ -32,7 +32,7 @@ jobs:
         run: ./scripts/cibuild
 
-      - uses: aws-actions/configure-aws-credentials@v1
+      - uses: aws-actions/configure-aws-credentials@v4
```
This is not the culprit, but updating the version to the latest is nice and does not break anything.

aaronxsu commented on deployment/aws-batch/docker-compose.yml:
```diff
@@ -1,11 +1,12 @@
+    environment:
+      - AWS_DEFAULT_REGION
+      - AWS_ACCESS_KEY_ID
+      - AWS_SECRET_ACCESS_KEY
```
Define in `docker-compose` that the above two vars should be read in from env.

aaronxsu commented on scripts/infra:
```diff
@@ -27,7 +27,9 @@ function update_batch_definitions() {
 
     ENVIRONMENT="${ENVIRONMENT:-staging}"
-    BATCH_ANALYSIS_JOB_NAME_REVISION=$(docker compose run --rm \
+    BATCH_ANALYSIS_JOB_NAME_REVISION=$(AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID} \
```
There might be a better way to pass them in, but we don't use a dotenv file as the env file in `docker-compose`, so this is the most straightforward way to do it I think. Please suggest if there is a more ideal solution!

KlaasH reviewed (approved):
This looks good, and it's nice that it's this simple.  As noted in response to your comment in `scripts/infra`, I think the change there isn't needed.

With that change backed out (and assuming an updated test branch still passes), this will be good to go :+1:

aaronxsu commented:
Thanks for the review. I removed it and pushed up a `test` branch here https://github.com/azavea/pfb-network-connectivity/actions/runs/13395094103. It went through. I updated this branch and will merge after CI.
